### PR TITLE
SNOW-1023214: Support date_part argument in last_day

### PR DESCRIPTION
**Pull request by sfc-gh-nkrishna**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   SNOW-1023214

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   This PR adds support for the optional `date_part` argument for the Snowflake function `last_day`. See the SQL documentation here: https://docs.snowflake.com/en/sql-reference/functions/last_day

### CODE REVIEW

File: src/snowflake/snowpark/functions.py [hunk 1]
```diff
@@ -3244,7 +3244,7 @@ def hour(e: ColumnOrName) -> Column:
     return builtin("hour")(c)
 
 
-def last_day(e: ColumnOrName) -> Column:
+def last_day(expr: ColumnOrName, part: Optional[ColumnOrName] = "MONTH") -> Column:
```
Comment (sfc-gh-stan) [inline src/snowflake/snowpark/functions.py:3247]: Let's add an `Args` section in the docstring now that there are two parameters.

File: CHANGELOG.md [hunk 2]
```diff
@@ -50,6 +50,7 @@
 - When parsing data types during a `to_pandas` operation, we rely on GS precision value to fix precision issues for large integer values. This may affect users where a column that was earlier returned as `int8` gets returned as `int64`. Users can fix this by explicitly specifying precision values for their return column.
 - Aligned behavior for `Session.call` in case of table stored procedures where running `Session.call` would not trigger stored procedure unless a `collect()` operation was performed.
 - `StoredProcedureRegistration` will now automatically add `snowflake-snowpark-python` as a package dependency. The added dependency will be on the client's local version of the library and an error is thrown if the server cannot support that version.
+- Support for optional `date_part` argument in Snowpark function `last_day`
```
Comment (sfc-gh-stan) [inline CHANGELOG.md:53]:
```suggestion
- Added support for an optional `date_part` argument in function `last_day`
```?

File: src/snowflake/snowpark/functions.py (hunk 1 above)
Comment (sfc-gh-nkrishna) [reply src/snowflake/snowpark/functions.py:3247]: Done!

**Review by sfc-gh-stan — APPROVED** [review]

**Review by sfc-gh-evandenberg — APPROVED** [review]
LGTM

**Pull request merged**